### PR TITLE
Wrappers for methods in version 2.4.13 of HyPHY

### PR DESCRIPTION
**Pull request by davebx**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection

FOR REVIEWER:
* [ ] .shed.yml file ok
    - [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
* [ ] Indentation is correct (4 spaces)
* [ ] Tool version/build ok
* [ ] `<command/>`
  - [ ] Text parameters, input and output files `'single quoted'`
  - [ ] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [ ] Tests
  - [ ] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [ ] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

### CODE REVIEW

File: tools/hyphy/hyphy_absrel.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0"?>
+<tool id="hyphy_absrel" name="HyPhy-aBSREL" version="@VERSION@+galaxy0">
+    <description>: adaptive Branch Site REL</description>
```
Comment (bgruening) [inline tools/hyphy/hyphy_absrel.xml]: the colon is unusual, I don't mind to leaf it in, just wanted to point it out :)

File: tools/hyphy/hyphy_absrel.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        </param>
+        <param name="branches" type="select" label="Set of branches to test">
+            <option value="1">All branches</option>
+            <option value="2">Internal branches</option>
+            <option value="3">Leaf branches</option>
+            <option value="4">Unlabeled branches</option>
+        </param>
+    </inputs>
+    <outputs>
+        <data name="absrel_log" format="txt" />
```
Comment (bgruening) [inline tools/hyphy/hyphy_absrel.xml]: Should this be optional?

File: tools/hyphy/hyphy_busted.xml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    echo `pwd`/busted_input.fa >> tool_params &&
+    echo `pwd`/busted_input.nhx >> tool_params &&
+    echo '$branches' >> tool_params &&
+    @HYPHY_INVOCATION@ \$HYPHY_LIB/TemplateBatchFiles/SelectionAnalyses/BUSTED.bf > '$busted_log'
+    ]]>
+    </command>
+    <inputs>
+        <param name="input_file" type="data" format="fasta" label="Input FASTA file" />
+        <param name="input_nhx" type="data" format="nhx" label="Input newick file" />
+        <param name="gencodeid" type="select" label="Genetic code">
```
Comment (bgruening) [inline tools/hyphy/hyphy_busted.xml]: Should this be also a macro?

File: tools/hyphy/hyphy_busted.xml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            <option value="5">Invertebrate mitochondrial DNA code</option>
+            <option value="6">Ciliate, Dasycladacean and Hexamita Nuclear code</option>
+            <option value="7">Echinoderm mitochondrial DNA code</option>
+            <option value="8">Euplotid Nuclear code</option>
+            <option value="9">Alternative Yeast Nuclear code</option>
+            <option value="10">Ascidian mitochondrial DNA code</option>
+            <option value="11">Flatworm mitochondrial DNA code</option>
+            <option value="12">Blepharisma Nuclear code</option>
+        </param>
+        <param name="branches" type="select" label="Set of branches to test">
```
Comment (bgruening) [inline tools/hyphy/hyphy_busted.xml]: And this as well?

File: tools/hyphy/hyphy_busted.xml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+<tool id="hyphy_busted" name="HyPhy-BUSTED" version="@VERSION@+galaxy0">
+    <description>: Branch-site Unrestricted Statistical Test for Episodic Diversification</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <version_command>HYPHYMP --version | tail -n 1 | awk '{print $1}'</version_command>
+    <command detect_errors="exit_code">
+    <![CDATA[
+    ln -s '$input_file' busted_input.fa &&
```
Comment (bgruening) [inline tools/hyphy/hyphy_busted.xml]: This could also be macrofied.

File: tools/hyphy/hyphy_busted.xml (hunk 4 above)
Comment (davebx) [reply tools/hyphy/hyphy_busted.xml]: For some reason, github is refusing to hide this as resolved.

File: tools/hyphy/hyphy_absrel.xml (hunk 2 above)
Comment (davebx) [reply tools/hyphy/hyphy_absrel.xml]: Yeah, I guess that makes sense.

File: tools/hyphy/hyphy_absrel.xml (hunk 1 above)
Comment (davebx) [reply tools/hyphy/hyphy_absrel.xml]: Yeah, I think I'll remove them.

**Comment by davebx** [comment]
@bgruening from what I understand, these tools prepare data for various visualizations. The conclusion I'd jump to first is that anyone using these tools either knows how to convert it to the next tool's format, or the next tool already comprehends HyPhy JSON output.

File: tools/hyphy/hyphy_absrel.xml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <expand macro="version_command" />
+    <command detect_errors="exit_code">
+    <![CDATA[
+    ln -s '$input_file' absrel_input.fa &&
+    ln -s '$input_nhx' absrel_input.nhx &&
+    echo $gencodeid > tool_params &&
```
Comment (bgruening) [inline tools/hyphy/hyphy_absrel.xml]: You could also use the config-file apprach here isn't it?

Comment (davebx) [reply tools/hyphy/hyphy_absrel.xml]: I did have that initially, but some complication I don't clearly recall made this way work better.

**Comment by bgruening** [comment]
Merci @davebx!

**Pull request merged**